lib/backend/opendistro_security.js

```diff
@@ -72,7 +72,7 @@ export default class SecurityBackend {
         }
     }
 
-    async authenticateWithHeader(headerName, headerValue) {
+    async authenticateWithHeader(headerName, headerValue, whitelistedHeaders) {
```

Comment:
is whitelistedHeaders a list (headerName, value pair) ?
Can we change function signature and allow all whitelisted header which we any way doing for all header types except Basic on line 88 ?

Reply:
No, we need to use session credentials from the cookie instead of headers. Also, we should scope this change to only basic auth for now.